shipyard-controller/handler/eventhandler.go

```diff
@@ -114,15 +115,17 @@ func (eh *EventHandler) HandleEvent(c *gin.Context) {
 		})
 	}
 
-	err := eh.ShipyardController.HandleIncomingEvent(*event)
-	if err != nil {
-		if err == errNoMatchingEvent {
-			SetBadRequestErrorResponse(err, c)
-		} else {
-			SetInternalServerErrorResponse(err, c)
+	go func() {
+		err := eh.ShipyardController.HandleIncomingEvent(*event)
+		if err != nil {
+			if err == errNoMatchingEvent {
+				log.Infof("no matching event found for event %s", event.ID)
+			} else {
+				log.Errorf("error processing request: %s", err.Error())
+			}
+			return
```

Comment:
I wonder if this is the right place for creating a new goroutine. 

I'd much rather do it as late as possible in the request-response cycle, e.g. here (`shipyardcontroller.go`)
```
	switch statusType {
	case string(common.TriggeredEvent):
		return sc.handleTriggeredEvent(event)
	case string(common.StartedEvent):
		return sc.handleStartedEvent(event)
	case string(common.FinishedEvent):
		return sc.handleFinishedEvent(event)
	default:
		return nil
	}
```

Reply:
agreed, will change that